Source/DiabloUI/settingsmenu.cpp

```diff
@@ -347,7 +347,8 @@ void UiSettingsMenu()
 		vecDialogItems.push_back(std::make_unique<UiListItem>("", static_cast<int>(SpecialMenuEntry::None), UiFlags::ElementDisabled));
 		vecDialogItems.push_back(std::make_unique<UiListItem>(_("Previous Menu"), static_cast<int>(SpecialMenuEntry::PreviousMenu), UiFlags::ColorUiGold));
 
-		vecDialog.push_back(std::make_unique<UiList>(vecDialogItems, rectList.size.height / 26, rectList.position.x, rectList.position.y, rectList.size.width, 26, UiFlags::FontSize24 | UiFlags::AlignCenter));
+		size_t maxViewportSize = rectList.size.height / 26;
+		vecDialog.push_back(std::make_unique<UiList>(vecDialogItems, std::min<size_t>(maxViewportSize, vecDialogItems.size()), rectList.position.x, rectList.position.y, rectList.size.width, 26, UiFlags::FontSize24 | UiFlags::AlignCenter));

```

Comment:
There's one other call that creates a UiList with fewer than the viewportSize items in selhero.cpp, could potentially put the std::min call/logic in the constructor for UiList itself.